protocol-units/da/movement/protocol/client/src/http2.rs

```diff
@@ -8,7 +10,20 @@ pub struct Http2 {
 impl Http2 {
 	/// Connects to a light node service using the given connection string.
 	pub async fn connect(connection_string: &str) -> Result<Self, anyhow::Error> {
-		let client = LightNodeServiceClient::connect(connection_string.to_string()).await?;
+		let endpoint = Channel::from_shared(connection_string.to_string())?;
+
+		// Dynamically configure TLS based on the scheme (http or https)
+		let endpoint = if connection_string.starts_with("https://") {
```

Comment:
It would be nice to start using the `Url` domain type `tonic::transport`, as it has parsing implemented via `FromStr`, so you don't have to do naive string matching here.

Reply:
Do we do it now, or you can open an issue and work on i later?